### PR TITLE
Profiling tool: Add support for health check.

### PR DESCRIPTION
**Pull request by nartal1**
This PR adds these features:
 list Failed - Jobs, Stages and Tasks
 list - RemovedBlockManager()
 list - RemovedExecutors()
 list - PossibleUnsupportedSQLPlan()

This PR is mainly to include the health related metrics/information of the cluster. 
Added new eventlogs which have failed stages, tasks and exectuors removed.
Added tests to validate the API's.

### CODE REVIEW

**Comment by nartal1** [comment]
build

File: tools/README.md [hunk 1]
```diff
@@ -334,6 +334,35 @@ Shuffle Skew Check: (When task's Shuffle Read Size > 3 * Avg Stage-level size)
 |1       |2      |0             |2224  |1      |222.22         |8.8           |3333.33          |111.11          |0.01            |false     |ExceptionFailure(ai.rapids.cudf.CudfException,cuDF failure at: /dddd/xxxxxxx/ccccc/bbbbbbbbb/aaaaaaa|
 +--------+-------+--------------+------+-------+---------------+--------------+-----------------+----------------+----------------+----------+----------------------------------------------------------------------------------------------------+
 ```
+#### C. Health Check
+
+```
+Application application_1616746343401_0025 (index=1) failed tasks:
++-------+--------------+------+-------+------------------------------------+
+|stageId|stageAttemptId|taskId|attempt|endReason_first36char               |
```
Comment (tgravescs) [inline tools/README.md:342]: any reason to use 36 char here and 100 below? seems like we could use 100 in both.  Maybe put ... at the end of hte string if its longer.

File: tools/src/main/scala/com/nvidia/spark/rapids/tool/profiling/HealthCheck.scala [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+import com.nvidia.spark.rapids.tool.ToolTextFileWriter
+import scala.collection.mutable.ArrayBuffer
+
+import org.apache.spark.sql.rapids.tool.profiling.ApplicationInfo
+
+/**
+ * HealthCheck defined health check rules
+ */
+class HealthCheck(apps: ArrayBuffer[ApplicationInfo], textFileWriter:ToolTextFileWriter){
```
Comment (tgravescs) [inline tools/src/main/scala/com/nvidia/spark/rapids/tool/profiling/HealthCheck.scala:27]: nit space after :

File: tools/src/main/scala/com/nvidia/spark/rapids/tool/profiling/HealthCheck.scala [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+  require(apps.nonEmpty)
+
+  // Function to list all failed tasks , stages and jobs.
+  def listFailedJobsStagesTasks(): Unit = {
+    for (app <- apps) {
+      // Look for failed tasks.
+      val tasksMessageHeader = s"Application ${app.appId} (index=${app.index}) failed tasks:\n"
```
Comment (tgravescs) [inline tools/src/main/scala/com/nvidia/spark/rapids/tool/profiling/HealthCheck.scala:35]: I think the rest of the headers leave off the appid and app index, we should be consistent about that.

File: tools/src/main/scala/org/apache/spark/sql/rapids/tool/profiling/ApplicationInfo.scala [hunk 4]
```diff
@@ -140,6 +140,9 @@ class ApplicationInfo(
   var taskGettingResult: ArrayBuffer[SparkListenerTaskGettingResult] =
     ArrayBuffer[SparkListenerTaskGettingResult]()
 
+  //Unsupported SQL plan
```
Comment (tgravescs) [inline tools/src/main/scala/org/apache/spark/sql/rapids/tool/profiling/ApplicationInfo.scala:143]: nit: space after //

File: tools/src/main/scala/org/apache/spark/sql/rapids/tool/profiling/ApplicationInfo.scala [hunk 5]
```diff
@@ -513,6 +520,15 @@ class ApplicationInfo(
       } else {
         logInfo("No Plan node accums Found. Create an empty Plan node accums DataFrame.")
       }
+
+      // For unsupportedSQLPlanDF
+      allDataFrames += (s"unsupportedSQLplan_$index" -> unsupportedSQLplan.toDF)
+      if (unsupportedSQLplan.nonEmpty) {
+        logInfo(s"Total ${unsupportedSQLplan.size} Plan node accums for appID=$appId")
```
Comment (tgravescs) [inline tools/src/main/scala/org/apache/spark/sql/rapids/tool/profiling/ApplicationInfo.scala:527]: "Plan node accums" is copy and paste from above, change it to something like unsupported ops

File: tools/src/test/scala/com/nvidia/spark/rapids/tool/profiling/HealthCheckSuite.scala [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids.tool.profiling
+
+import java.io.File
+
+import com.nvidia.spark.rapids.tool.ToolTestUtils
+import org.scalatest.FunSuite
+import scala.collection.mutable.ArrayBuffer
```
Comment (tgravescs) [inline tools/src/test/scala/com/nvidia/spark/rapids/tool/profiling/HealthCheckSuite.scala:23]: nit: scala should go after java and  before this section

**Review by tgravescs — COMMENTED** [review]
you can ignore the nits if you want

**Review by tgravescs — APPROVED** [review]
Actually nothing major here so I'm going to merge and lets follow up with the first 36 char comment and the fixing the log statement in separate one.

**Pull request merged**